src/lib/components/dropdown/utils.js

```diff
@@ -0,0 +1,72 @@
+import { css } from 'styled-components';
+
+const getBorderCollisionDetection = (isItem, triggerSize, size) => {
+  let isTopHit = false
+  let isRightHit = false
+  let isBottomHit = false
+  let isLeftHit = false
+
+  if (size && triggerSize && triggerSize.top - size.height <= 0) {
+    isTopHit = true
+  }
+  if (size && triggerSize && triggerSize.right + triggerSize.width > window.innerWidth) {
+    isRightHit = true
+  }
+  if (size && triggerSize && triggerSize.top + size.height >= window.innerHeight) {
+    isBottomHit = true
+  }
+  if (size && triggerSize && triggerSize.left - triggerSize.width <= 0) {
+    isLeftHit = true
+  }
+
+  return {
+    isTopHit,
+    isRightHit,
+    isBottomHit,
+    isLeftHit
+  }
+}
+
+export const getPositionDropdownMenu = ({ isItem, triggerSize, size, nbItems, itemIndex }) => {
+  const { isTopHit, isRightHit, isBottomHit, isLeftHit } = getBorderCollisionDetection(isItem, triggerSize, size)
+
+  if (isItem) {
+    // Check collision for dropdown acting as an item
+    if (isRightHit && isBottomHit) {
+      return css`
+          right: ${size.width}px;
+          bottom: ${-triggerSize.height * (itemIndex - nbItems + 1)}px;
```

Comment:
@JBWatenbergScality This one is a bit tricky to handle because indeed if an item have a height superior to other item, the new menu generated will display with a wrong offset. The problem I have to get each item size of the previous menu to place the new menu correctly but it will add a lot of complexity to the code. I can figure out a way to implement that in a simple way, do you have any suggestions ?